docs/fundamentals/code-analysis/configuration-files.md

```diff
@@ -0,0 +1,123 @@
+
+You can apply EditorConfig file conventions to a folder, a project, or an entire repo by placing the file in the corresponding directory. These options will be applied when executing the analysis at build time as well as inside Visual Studio.
+
+If you have an existing *.editorconfig* file for editor settings such as indent size or whether to trim trailing whitespace, you can place your code analysis configuration options in the same file.
+
+> [!TIP]
+> Visual Studio provides an *.editorconfig* item template that makes is easy to add one of these files to your project. For more information, see [Add an EditorConfig file to a project](/visualstudio/ide/create-portable-custom-editor-options#add-an-editorconfig-file-to-a-project).
+
+### Example
+Following is an example EditorConfig file to configure options and rule severity:
+
+```ini
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
+root = true
+
+# C# files
+[*.cs]
+
+#### Core EditorConfig Options ####
+
+# Indentation and spacing
+indent_size = 4
+indent_style = space
+tab_width = 4
+
+#### .NET Coding Conventions ####
+
+# this. and Me. preferences
+dotnet_style_qualification_for_method = true:warning
+
+#### Diagnostic configuration ####
+
+# CA1000: Do not declare static members on generic types
+dotnet_diagnostic.CA1000.severity = warning
+```
+
+## Global AnalyzerConfig
+
+Starting with .NET5 SDK (which is supported starting with Visual Studio 2019 16.8), you can also configure analyzer options with Global AnalyzerConfig files. These files are used to provide **options that apply to all the source files in the project**, regardless of the actual file names or file paths of these source files.
+
+Unlike [EditorConfig](#editorconfig) files, global config files are not designed to configure editor style settings for IDEs, such as indent size or whether to trim trailing whitespace. Instead, they are designed purely for specifying project level analyzer configuration options.
+
+### Naming
+
+Unlike EditorConfig files, which must be named `.editorconfig`, global config files do not need to have a specific name or extension. However, if you name these files as `.globalconfig` then they will be implicitly applied to all the C# and Visual Basic projects within the current folder, including subfolders. Otherwise, you will need to explicitly add the following MSBuild item group to your MSBuild project file:
+
+```xml
+<ItemGroup>
+  <GlobalAnalyzerConfigFiles Include="<%path_to_global_analyzer_config%>" />
+</ItemGroup>
+```
+
+### Format
+
+Unlike EditorConfig files, which must have section headers, such as `[*.cs]`, to identify the applicable files and folders, Global AnalyzerConfig do not have section headers. Instead, they require a top level entry of the form `is_global = true`. This indicates that all the options in the file apply globally to the entire project. For example,
```

Comment:
What is the behavior of is_global is set to false or omitted?

Reply:
It has no effect - compiler only searches for presence of `is_global = true`

Reply:
I think we need to state the behavior if it's omitted. E.g. does it just apply to files in the current folder?

Reply:
And, if you name the file .globalconfig, do you still need it?

Reply:
> I think we need to state the behavior if it's omitted. E.g. does it just apply to files in the current folder?

If it is omitted, compiler just treats it as a normal editorconfig, so it will basically apply all settings within section headers to files under it.

> And, if you name the file .globalconfig, do you still need it?

Yes, that is the current design. `is_global = true` is still needed.

Reply:
In that case, this line seems a little misleading?

"if you name these files as `.globalconfig` then they will be implicitly applied to all the C# and Visual Basic projects within the current folder, including subfolders"